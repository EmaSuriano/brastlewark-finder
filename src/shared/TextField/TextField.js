import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Input = styled.input`
  padding: 0.5em;
  margin: 0.5em;
  color: palevioletred;
  background: papayawhip;
  border: none;
  border-radius: 3px;
`;

const TextField = ({ value, onChange }) => (
  <Input
    type="text"
    onChange={({ target: { value } }) => onChange(value)}
    value={value}
  />
);

TextField.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func,
};

export default TextField;
