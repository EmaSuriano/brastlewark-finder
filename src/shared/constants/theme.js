export default {
  background: '#4F5D75',
  light: '#FFFFFF',
  disabled: '#BFC0C0',
  accent: '#EF8354',
  secondary: '#2D3142',
};
