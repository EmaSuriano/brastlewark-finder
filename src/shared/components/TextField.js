import styled from 'styled-components';
import { ThemePropTypes } from 'shared/constants/theme';

const TextField = styled.input.attrs({
  type: 'text',
})`
  padding: 0.5em;
  margin: 0.5em;
  color: ${props => props.theme.secondary};
  background: ${props => props.theme.light};
  border: none;
  border-radius: 3px;
`;

TextField.propTypes = {
  theme: ThemePropTypes,
};

TextField.displayName = 'TextField';

export default TextField;
