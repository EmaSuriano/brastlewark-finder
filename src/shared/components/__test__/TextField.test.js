import React from 'react';
import { shallow } from 'enzyme';
import theme from 'shared/constants/theme';
import TextField from '../TextField';

describe('<TextField />', () => {
  it('render', () => {
    expect(shallow(<TextField theme={theme} />)).toMatchSnapshot();
  });
});
