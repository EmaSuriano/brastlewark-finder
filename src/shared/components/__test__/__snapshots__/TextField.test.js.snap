// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> render 1`] = `
.c0 {
  padding: 0.5em;
  margin: 0.5em;
  color: #2D3142;
  background: #FFFFFF;
  border: none;
  border-radius: 3px;
}

<input
  className="c0"
  type="text"
/>
`;
